<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Installing-Text-Fabric" data-toc-modified-id="Installing-Text-Fabric-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Installing Text-Fabric</a></span></li><li><span><a href="#Checking-our-results-with-GREP" data-toc-modified-id="Checking-our-results-with-GREP-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Checking our results with GREP</a></span></li><li><span><a href="#Reflection-on-GREP" data-toc-modified-id="Reflection-on-GREP-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Reflection on GREP</a></span></li></ul></div>

# Primer 3

This is the third in a series of primers on the use of [Text-Fabric](https://github.com/Dans-labs/text-fabric) for analyzing proto-cuneiform. This primer continues with our exploration of subcases in proto-cuneiform, but focuses the use of queries within Text-Fabric and verifying the results of a TF query, using regular expressions in the source text.

Primer 1 is available [here](primer1.ipnynb), and Primer 2 [here](primer2.ipynb), and there is also a set of tutorials on the use of Text-Fabric to analyze proto-cuneiform are available [here](https://github.com/Nino-cunei/tutorials/blob/master/start.ipynb).

In this primer we are going to focus on a tablet in which individual portions of meat and fish are being distributed to a group of elites personnel or office-holders on a specific occasion: [P005213](https://cdli.ucla.edu/P005213), which was published as MSVO 1, 146. Two orthographies are characteristic of this text, the occurrence of SZITA~a1 or related signs for elite distributions such as |SILA3~axKU6~a|, typically alongside a numeral in a subcase, and |ZATU659x1(N01)| alongside a job title in the superordinate case.

## Installing Text-Fabric

See [primer1](primer1.ipynb)

###### Execute: If all this is done, the following cells can be executed.

In [1]:
import os, sys, collections
from IPython.display import display, Markdown
from tf.extra.cunei import Cunei

In [2]:
LOC = ('~/github', 'Nino-cunei/uruk', 'primer3')
A = Cunei(*LOC)
A.api.makeAvailableIn(globals())

Found 2095 ideograph linearts
Found 2724 tablet linearts
Found 5495 tablet photos


**Documentation:** <a target="_blank" href="https://github.com/Nino-cunei/uruk/blob/master/docs/about.md" title="provenance of this corpus">Uruk IV-III (v1.0)</a> <a target="_blank" href="https://github.com/Nino-cunei/uruk/blob/master/docs/transcription.md" title="feature documentation">Feature docs</a> <a target="_blank" href="https://dans-labs.github.io/text-fabric/Api/Cunei/" title="cunei api documentation">Cunei API</a> <a target="_blank" href="https://dans-labs.github.io/text-fabric/Api/General/" title="text-fabric-api">Text-Fabric API 4.1.2</a> <a target="_blank" href="https://dans-labs.github.io/text-fabric/Api/General/#search-templates" title="Search Templates Introduction and Reference">Search Reference</a>


This notebook online:
<a target="_blank" href="http://nbviewer.jupyter.org/github/Nino-cunei/tutorials/blob/master/bits-and-pieces/primer3.ipynb">NBViewer</a>
<a target="_blank" href="https://github.com/Nino-cunei/tutorials/blob/master/bits-and-pieces/primer3.ipynb">GitHub</a>


In [3]:
def dm(markdown): display(Markdown(markdown))

So let's pull up our text in Text-Fabric and then walk through how to create a query in Text-Fabric that brings up all examples with the same key orthographies.

In [4]:
pNum = 'P005213'
query = f'''
tablet catalogId={pNum}
'''
results = A.search(query)
tablet = results[0][0]
A.lineart(tablet, width=400)
A.getSource(tablet)

1 result


['&P005213 = MSVO 1, 146',
 '#atf: lang qpc ',
 '@obverse ',
 '@column 1 ',
 '1.a. [...] , [...] ',
 '1.b. [...] , [...] X ',
 '2.a. [...] 2(N01)# , SZITA~a1 ',
 '2.b. 1(N01) , |SILA3~axKU6~a| ',
 '3.a. 1(N08) , [...] X ',
 '3.b. 2(N01) , SZITA~a1 ',
 '4.a. |ZATU659x1(N01)|# , SAG# [...] ',
 '4.b. 1(N01) , SZITA~a1 ',
 '5.a. |ZATU659x1(N01)|# , SIG7# SAL#? X ',
 '5.b. [...] , X ',
 '6.a. |ZATU659x1(N01)| , TU~b HI SZA3~a2 ',
 '6.b. 1(N01) , SZITA~a1 ',
 '7.a. |ZATU659x1(N01)| , AMA~a AGAR2 ',
 '7.b. 1(N01) , SZITA~a1 ',
 '8.a. |ZATU659x1(N01)| , AN NIM~b3 ',
 '8.b. 1(N01) , SZITA~a1 ',
 '9.a. 2(N01) , GAR |NI~a.RU| ME~a ',
 '9.b1. 1(N01) , KASZ~a ',
 '9.b2. 1(N01) , SZITA~a1 ',
 '@column 2 ',
 '1.a. [...] , ENGIZ# SZAGAN# ',
 '1.b. 1(N01) , SZITA~a1 ',
 '2.a. 1(N08) , EN~a UMUN2 ABGAL ',
 '2.b. 1(N01) , SZITA~a1 ',
 '3.a. |ZATU659x1(N01)|# , SANGA~a# |NI~a.RU|# ',
 '3.b. 1(N01) , SZITA~a1 ',
 '4.a. 1(N08) , SANGA~a AB~a ',
 '4.b. 1(N01) , SZITA~a1 ',
 '5.a. 1(N08) , [...] IB~a# ',
 '5.

Let's start by bring up one line, namely the second line in the second column, and once we have described its specific form, we can try and construct a query that will bring up all similar examples.

First, a simple query that directly refers to this line. 

In [5]:
query = f'''
tablet catalogId={pNum}
  face type=obverse
    column number=2
        line number=2
'''
results = A.search(query)
A.table(results)

1 result


n | tablet | face | column | line
--- | --- | --- | --- | ---
1|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|face obverse|column 2|line 2

In [6]:
oneLine = results[0][3]
A.pretty(oneLine, withNodes=True, lineart=True)

Here we can see nine nodes in Text-Fabric, each marked by a grey box with a six digit number. The top level node belongs to the line (252933), each case has its own node (2a = 167110, and 2b = 167111) and so on.

The distinctive orthography we want to look at first is the occurrence of SZITA~a1 in a subcase (2b), as a qualification of the official or individual listed in the other half of the line (2a).

Now let's formulate a query that looks for any occurrence of SZITA~1 at sublevel 1 or lower.

In [7]:
query = '''
case depth=1
  sign grapheme=SZITA variant=a1
'''
results = A.search(query)
A.table(results, withNodes=True, lineart=False)

78 results


n | case | sign
--- | --- | ---
1|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P002113" title="to CDLI main page for this tablet">case 1a</a> *159945* |SZITA~a1 *5882* 
2|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P002179" title="to CDLI main page for this tablet">case 1a</a> *159964* |SZITA~a1 *6287* 
3|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P002179" title="to CDLI main page for this tablet">case 3c</a> *159972* |SZITA~a1 *6316* 
4|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P002179" title="to CDLI main page for this tablet">case 6d</a> *159993* |SZITA~a1 *6390* 
5|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P002179" title="to CDLI main page for this tablet">case 9c</a> *160005* |SZITA~a1 *6437* 
6|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P002399" title="to CDLI main page for this tablet">case 2b</a> *160107* |SZITA~a1 *8262* 
7|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P002522" title="to CDLI main page for this tablet">case 1c</a> *160130* |SZITA~a1 *8808* 
8|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003654" title="to CDLI main page for this tablet">case 6b</a> *160467* |SZITA~a1 *12531* 
9|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004442" title="to CDLI main page for this tablet">case 3b</a> *160666* |SZITA~a1 *14742* 
10|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004442" title="to CDLI main page for this tablet">case 1a</a> *160667* |SZITA~a1 *14749* 
11|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004442" title="to CDLI main page for this tablet">case 1b</a> *160670* |SZITA~a1 *14754* 
12|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004676" title="to CDLI main page for this tablet">case 3b</a> *160770* |SZITA~a1 *16619* 
13|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004684" title="to CDLI main page for this tablet">case 2a</a> *160808* |SZITA~a1 *16907* 
14|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004884" title="to CDLI main page for this tablet">case 1a</a> *161115* |SZITA~a1 *20930* 
15|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P212396" title="to CDLI main page for this tablet">case 2a</a> *161251* |SZITA~a1 *22873* 
16|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P325063" title="to CDLI main page for this tablet">case 5a</a> *162089* |SZITA~a1 *52820* 
17|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P325063" title="to CDLI main page for this tablet">case 1a</a> *162095* |SZITA~a1 *52833* 
18|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P325074" title="to CDLI main page for this tablet">case 3a</a> *162265* |SZITA~a1 *53831* 
19|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P325069" title="to CDLI main page for this tablet">case 1a</a> *162277* |SZITA~a1 *53978* 
20|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P325069" title="to CDLI main page for this tablet">case 3a</a> *162279* |SZITA~a1 *53985* 
21|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P002425" title="to CDLI main page for this tablet">case 3a</a> *163664* |SZITA~a1 *68284* 
22|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P002387" title="to CDLI main page for this tablet">case 2a</a> *163674* |SZITA~a1 *68445* 
23|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004499" title="to CDLI main page for this tablet">case 1b</a> *164005* |SZITA~a1 *71918* 
24|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004499" title="to CDLI main page for this tablet">case 2b</a> *164007* |SZITA~a1 *71923* 
25|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004499" title="to CDLI main page for this tablet">case 3b</a> *164009* |SZITA~a1 *71928* 
26|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004499" title="to CDLI main page for this tablet">case 10b</a> *164011* |SZITA~a1 *71989* 
27|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004313" title="to CDLI main page for this tablet">case 1a</a> *164022* |SZITA~a1 *72342* 
28|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003807" title="to CDLI main page for this tablet">case 1c</a> *164101* |SZITA~a1 *72932* 
29|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003617" title="to CDLI main page for this tablet">case 2b</a> *164176* |SZITA~a1 *74294* 
30|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003655" title="to CDLI main page for this tablet">case 1a</a> *164229* |SZITA~a1 *74852* 
31|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004299" title="to CDLI main page for this tablet">case 2a</a> *164363* |SZITA~a1 *77575* 
32|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004501" title="to CDLI main page for this tablet">case 1a</a> *164422* |SZITA~a1 *78954* 
33|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004501" title="to CDLI main page for this tablet">case 1b</a> *164423* |SZITA~a1 *78956* 
34|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004501" title="to CDLI main page for this tablet">case 2b</a> *164425* |SZITA~a1 *78963* 
35|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004501" title="to CDLI main page for this tablet">case 3b</a> *164427* |SZITA~a1 *78968* 
36|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004501" title="to CDLI main page for this tablet">case 4b</a> *164429* |SZITA~a1 *78973* 
37|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004510" title="to CDLI main page for this tablet">case 2a</a> *164441* |SZITA~a1 *79080* 
38|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P006253" title="to CDLI main page for this tablet">case 6a</a> *165953* |SZITA~a1 *90035* 
39|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P006286" title="to CDLI main page for this tablet">case 6a</a> *166128* |SZITA~a1 *91971* 
40|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P006286" title="to CDLI main page for this tablet">case 6b</a> *166129* |SZITA~a1 *91977* 
41|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005151" title="to CDLI main page for this tablet">case 3b</a> *166743* |SZITA~a1 *97957* 
42|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005175" title="to CDLI main page for this tablet">case 2a</a> *166924* |SZITA~a1 *99477* 
43|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005178" title="to CDLI main page for this tablet">case 7a</a> *166958* |SZITA~a1 *99707* 
44|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">case 2a</a> *167090* |SZITA~a1 *100889* 
45|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">case 3b</a> *167093* |SZITA~a1 *100897* 
46|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">case 4b</a> *167095* |SZITA~a1 *100903* 
47|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">case 6b</a> *167099* |SZITA~a1 *100917* 
48|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">case 7b</a> *167101* |SZITA~a1 *100923* 
49|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">case 8b</a> *167103* |SZITA~a1 *100929* 
50|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">case 9b</a> *167105* |SZITA~a1 *100938* 
51|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">case 1b</a> *167109* |SZITA~a1 *100943* 
52|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">case 2b</a> *167111* |SZITA~a1 *100949* 
53|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">case 3b</a> *167113* |SZITA~a1 *100956* 
54|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">case 4b</a> *167115* |SZITA~a1 *100961* 
55|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">case 5b</a> *167117* |SZITA~a1 *100966* 
56|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">case 6b</a> *167119* |SZITA~a1 *100971* 
57|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">case 1c</a> *167135* |SZITA~a1 *101052* 
58|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">case 2c</a> *167140* |SZITA~a1 *101062* 
59|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">case 3c</a> *167145* |SZITA~a1 *101071* 
60|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">case 4c</a> *167148* |SZITA~a1 *101079* 
61|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">case 5b</a> *167150* |SZITA~a1 *101087* 
62|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">case 6b</a> *167152* |SZITA~a1 *101091* 
63|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005215" title="to CDLI main page for this tablet">case 1b</a> *167154* |SZITA~a1 *101137* 
64|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005217" title="to CDLI main page for this tablet">case 1b</a> *167156* |SZITA~a1 *101157* 
65|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005217" title="to CDLI main page for this tablet">case 3b</a> *167160* |SZITA~a1 *101169* 
66|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005300" title="to CDLI main page for this tablet">case 2a</a> *167488* |SZITA~a1 *103740* 
67|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005458" title="to CDLI main page for this tablet">case 4a</a> *167955* |SZITA~a1 *109529* 
68|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003499" title="to CDLI main page for this tablet">case 2a</a> *168153* |SZITA~a1 *111960* 
69|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003541" title="to CDLI main page for this tablet">case 1b</a> *168431* |SZITA~a1 *113331* 
70|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003593" title="to CDLI main page for this tablet">case 2a</a> *168537* |SZITA~a1 *114236* 
71|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003593" title="to CDLI main page for this tablet">case 3b</a> *168542* |SZITA~a1 *114248* 
72|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003808" title="to CDLI main page for this tablet">case 2b</a> *168685* |SZITA~a1 *114965* 
73|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003808" title="to CDLI main page for this tablet">case 3b</a> *168702* |SZITA~a1 *114999* 
74|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003808" title="to CDLI main page for this tablet">case 2a</a> *168707* |SZITA~a1 *115035* 
75|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003808" title="to CDLI main page for this tablet">case 2b</a> *168715* |SZITA~a1 *115046* 
76|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003808" title="to CDLI main page for this tablet">case 1b</a> *168725* |SZITA~a1 *115073* 
77|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003808" title="to CDLI main page for this tablet">case 2b</a> *168736* |SZITA~a1 *115108* 
78|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P002545" title="to CDLI main page for this tablet">case 1a</a> *169064* |SZITA~a1 *128468* 

So 78 hits in the corpus as a whole, but this simply tells us that SZITA~a1 is a frequent occupant of divided lines. Let's check some other levels and then, after that, try to limit the search to a specific tablet.

In [8]:
query = '''
case depth=2
  sign grapheme=SZITA variant=a1
'''
results = A.search(query)
A.table(results, withNodes=True, lineart=False)

12 results


n | case | sign
--- | --- | ---
1|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003654" title="to CDLI main page for this tablet">case 6b1</a> *160468* |SZITA~a1 *12531* 
2|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004442" title="to CDLI main page for this tablet">case 1a2</a> *160669* |SZITA~a1 *14749* 
3|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P004684" title="to CDLI main page for this tablet">case 2a3</a> *160811* |SZITA~a1 *16907* 
4|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P325074" title="to CDLI main page for this tablet">case 3a1</a> *162266* |SZITA~a1 *53831* 
5|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003655" title="to CDLI main page for this tablet">case 1a3</a> *164232* |SZITA~a1 *74852* 
6|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P006286" title="to CDLI main page for this tablet">case 6b2</a> *166131* |SZITA~a1 *91977* 
7|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005151" title="to CDLI main page for this tablet">case 3b1</a> *166744* |SZITA~a1 *97957* 
8|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">case 9b2</a> *167107* |SZITA~a1 *100938* 
9|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">case 1c1</a> *167136* |SZITA~a1 *101052* 
10|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">case 2c1</a> *167141* |SZITA~a1 *101062* 
11|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003808" title="to CDLI main page for this tablet">case 2b1</a> *168686* |SZITA~a1 *114965* 
12|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003808" title="to CDLI main page for this tablet">case 2a4</a> *168713* |SZITA~a1 *115035* 

In [9]:
query = '''
case depth=3
  sign grapheme=SZITA variant=a1
'''
results = A.search(query)
A.table(results, withNodes=True, lineart=False)

1 result


n | case | sign
--- | --- | ---
1|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003808" title="to CDLI main page for this tablet">case 2b1B</a> *168688* |SZITA~a1 *114965* 

In [10]:
query = '''
case depth=4
  sign grapheme=SZITA variant=a1
'''
results = A.search(query)
A.table(results, withNodes=True, lineart=False)

1 result


n | case | sign
--- | --- | ---
1|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P003808" title="to CDLI main page for this tablet">case 2b1B2</a> *168690* |SZITA~a1 *114965* 

Note in particular that our query with depth=3 is actually picking up the same single item at depth=4 because we are not focusing here on terminal subdivisions like we did in primer 2. This is made particularly visible by leaving the Text-Fabric node labels in, so as soon as we see that the sign in question is at node 114965, we know that this is the same proto-cuneiform sign in all four of these queries.

Now let's limit this query to the tablet that we are focusing on here, namely P005213 aka MSVO 1, 146.

In [11]:
query = f'''
tablet catalogId={pNum}
 case depth=1
   sign grapheme=SZITA variant=a1
'''
results = A.search(query)
A.table(results, withNodes=False, lineart=False)

13 results


n | tablet | case | sign
--- | --- | --- | ---
1|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2a|SZITA~a1
2|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1
3|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1
4|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1
5|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 7b|SZITA~a1
6|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 8b|SZITA~a1
7|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 9b|SZITA~a1
8|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 1b|SZITA~a1
9|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2b|SZITA~a1
10|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1
11|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1
12|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 5b|SZITA~a1
13|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1

There are thirteen occurrences of SZITA~a1 in P005213, which means that one-sixth (13/78) of all occurrences of SZITA~a1 in the entire corpus occur in this one tablet. 

The other orthography that plays a central role in P005213 is ZATU659x1(N01), which regularly occurs in the other half of the subdivided lines that include SZITA~a1 in the other case in the line.

In [12]:
query = f'''
tablet catalogId={pNum}
  case
    quad
      =: sign grapheme=ZATU659
      <: sign grapheme=N01 repeat=1
'''
results = A.search(query)
A.table(results, withNodes=False, lineart=False)

7 results


n | tablet | case | quad | sign | sign
--- | --- | --- | --- | --- | ---
1|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
2|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 5a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
3|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
4|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 7a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
5|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 8a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
6|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
7|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)

In [14]:
A.show(results, withNodes=True, start=1, end=1)


##### Tablet 1


In [16]:
exampleCase = 167112
A.pretty(exampleCase, withNodes=True)

In [19]:
q = 142753
s = 100950

print(f'slots of the quad: {L.d(q, otype="sign")}')

slots of the quad: [100950, 100951]


Up to this point, our queries have consisted of "atom" lines, which identify a particular node in the corpus. The preceding query, however, also includes "relation" lines, namely the two lines that begin with "=:" and "<:". The first of these, =:, means "=: left and right start at the same slot", while <: means "<: :>: adjacent before and after (with respect to the slots occupied: left ends immediately before right starts and vice versa)".

In [32]:
query = '''
tablet
  ln:line
  
cs1:case depth=1 terminal
  sign grapheme=SZITA variant=a1

cs2:case
  quad
    =: sign grapheme=ZATU659
    <: sign grapheme=N01 repeat=1

ln [[ cs1
ln [[ cs2
'''
results = A.search(query)
A.table(results, withNodes=False, lineart=False)

9 results


n | tablet | line | case | sign | case | quad | sign | sign
--- | --- | --- | --- | --- | --- | --- | --- | ---
1|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P002399" title="to CDLI main page for this tablet">tablet P002399</a>|line 2|case 2b|SZITA~a1|case 2a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
2|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|line 4|case 4b|SZITA~a1|case 4a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
3|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|line 6|case 6b|SZITA~a1|case 6a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
4|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|line 7|case 7b|SZITA~a1|case 7a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
5|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|line 8|case 8b|SZITA~a1|case 8a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
6|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|line 3|case 3b|SZITA~a1|case 3a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
7|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">tablet P005214</a>|line 4|case 4c|SZITA~a1|case 4a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
8|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">tablet P005214</a>|line 5|case 5b|SZITA~a1|case 5a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
9|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005215" title="to CDLI main page for this tablet">tablet P005215</a>|line 1|case 1b|SZITA~a1|case 1a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)

In [33]:
S.study(query)

   |     0.00s Feature overview: 28 for nodes; 4 for edges; 1 configs; 7 computed
   |     0.00s Checking search template ...
   |     0.00s Setting up search space for 8 objects ...
   |     0.73s Constraining search space with 9 relations ...
   |     0.74s Setting up retrieval plan ...
   |     0.74s Ready to deliver results from 42386 nodes
   |   Iterate over S.fetch() to get the results
   |   See S.showPlan() to interpret the results


In [35]:
S.showPlan()

   |       15s The results are connected to the original search template as follows:
   |    0     
   |    1 R0  tablet
   |    2 R1    ln:line
   |    3       
   |    4 R2  cs1:case depth=1 terminal
   |    5 R3    sign grapheme=SZITA variant=a1
   |    6     
   |    7 R4  cs2:case
   |    8 R5    quad
   |    9 R6      =: sign grapheme=ZATU659
   |   10 R7      <: sign grapheme=N01 repeat=1
   |   11     
   |   12     ln [[ cs1
   |   13     ln [[ cs2
   |   14     


In [36]:
query = '''
tablet
  line
    case depth=1 terminal
      sign grapheme=SZITA variant=a1
    case
      quad
        =: sign grapheme=ZATU659
        <: sign grapheme=N01 repeat=1
'''
results = A.search(query)
A.table(results, withNodes=False, lineart=False)

9 results


n | tablet | line | case | sign | case | quad | sign | sign
--- | --- | --- | --- | --- | --- | --- | --- | ---
1|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P002399" title="to CDLI main page for this tablet">tablet P002399</a>|line 2|case 2b|SZITA~a1|case 2a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
2|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|line 4|case 4b|SZITA~a1|case 4a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
3|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|line 6|case 6b|SZITA~a1|case 6a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
4|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|line 7|case 7b|SZITA~a1|case 7a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
5|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|line 8|case 8b|SZITA~a1|case 8a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
6|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|line 3|case 3b|SZITA~a1|case 3a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
7|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">tablet P005214</a>|line 4|case 4c|SZITA~a1|case 4a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
8|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">tablet P005214</a>|line 5|case 5b|SZITA~a1|case 5a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)
9|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005215" title="to CDLI main page for this tablet">tablet P005215</a>|line 1|case 1b|SZITA~a1|case 1a|&#124;ZATU659x1(N01)&#124;|ZATU659|1(N01)

In [37]:
S.study(query)

   |     0.00s Feature overview: 28 for nodes; 4 for edges; 1 configs; 7 computed
   |     0.00s Checking search template ...
   |     0.00s Setting up search space for 8 objects ...
   |     0.73s Constraining search space with 9 relations ...
   |     0.74s Setting up retrieval plan ...
   |     0.74s Ready to deliver results from 67 nodes
   |   Iterate over S.fetch() to get the results
   |   See S.showPlan() to interpret the results


In [39]:
S.showPlan(details=True)

   |   Search with 8 objects and 9 relations
   |   Results are instantiations of the following objects:
   |   node  0-tablet                            (     4   choices)
   |   node  1-line                              (     9   choices)
   |   node  2-case                              (     9   choices)
   |   node  3-sign                              (     9   choices)
   |   node  4-case                              (     9   choices)
   |   node  5-quad                              (     9   choices)
   |   node  6-sign                              (     9   choices)
   |   node  7-sign                              (     9   choices)
   |   Instantiations are computed along the following relations:
   |   node                      0-tablet        (     4   choices)
   |   edge  0-tablet        [[  1-line          (     2.2 choices)
   |   edge  1-line          [[  4-case          (     1.0 choices)
   |   edge  4-case          [[  5-quad          (     1.0 choices)
   |   edge  

In [27]:
A.show(results, lineart=False)


##### Tablet 1



##### Tablet 2



##### Tablet 3



##### Tablet 4


In [28]:
print(S.relationLegend)

                      = left equal to right (as node)
                      # left unequal to right (as node)
                      < left before right (in canonical node ordering)
                      > left after right (in canonical node ordering)
                     == left occupies same slots as right
                     && left has overlapping slots with right
                     ## left and right do not have the same slot set
                     || left and right do not have common slots
                     [[ left embeds right
                     ]] left embedded in right
                     << left completely before right
                     >> left completely after right
                     =: left and right start at the same slot
                     := left and right end at the same slot
                     :: left and right start and end at the same slot
                     <: left immediately before right
                     :> left immediately after right
   

In [40]:
F.otype.slotType

'sign'

This query gives us 95 occurrences in four distinct proto-cuneiform tablets: P005213, P005214, P005215 and P002399.

The two indented lines starting with "case" identify a distinct sign; 

=: left and right start at the same slot

<: :>: adjacent before and after (with respect to the slots occupied: left ends immediately before right starts and vice versa)

:= left and right end at the same slot

In [14]:
query = '''
tablet
  case depth=1 terminal
    sign grapheme=SZITA variant=a1
  case
    quad
      =: sign grapheme=ZATU659
      <: sign grapheme=N01 repeat=1
      :=
'''
results = A.search(query)
A.table(results, withNodes=False, lineart=False)

95 results


n | tablet | case | sign | case | quad | sign | sign
--- | --- | --- | --- | --- | --- | --- | ---
1|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 1b|SZITA~a1|case 3a||ZATU659x1(N01)||ZATU659|1(N01)
2|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2b|SZITA~a1|case 3a||ZATU659x1(N01)||ZATU659|1(N01)
3|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1|case 3a||ZATU659x1(N01)||ZATU659|1(N01)
4|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1|case 3a||ZATU659x1(N01)||ZATU659|1(N01)
5|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 5b|SZITA~a1|case 3a||ZATU659x1(N01)||ZATU659|1(N01)
6|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1|case 3a||ZATU659x1(N01)||ZATU659|1(N01)
7|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2a|SZITA~a1|case 3a||ZATU659x1(N01)||ZATU659|1(N01)
8|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1|case 3a||ZATU659x1(N01)||ZATU659|1(N01)
9|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1|case 3a||ZATU659x1(N01)||ZATU659|1(N01)
10|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1|case 3a||ZATU659x1(N01)||ZATU659|1(N01)
11|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 7b|SZITA~a1|case 3a||ZATU659x1(N01)||ZATU659|1(N01)
12|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 8b|SZITA~a1|case 3a||ZATU659x1(N01)||ZATU659|1(N01)
13|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 1b|SZITA~a1|case 6a||ZATU659x1(N01)||ZATU659|1(N01)
14|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2b|SZITA~a1|case 6a||ZATU659x1(N01)||ZATU659|1(N01)
15|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1|case 6a||ZATU659x1(N01)||ZATU659|1(N01)
16|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1|case 6a||ZATU659x1(N01)||ZATU659|1(N01)
17|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 5b|SZITA~a1|case 6a||ZATU659x1(N01)||ZATU659|1(N01)
18|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1|case 6a||ZATU659x1(N01)||ZATU659|1(N01)
19|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2a|SZITA~a1|case 6a||ZATU659x1(N01)||ZATU659|1(N01)
20|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1|case 6a||ZATU659x1(N01)||ZATU659|1(N01)
21|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1|case 6a||ZATU659x1(N01)||ZATU659|1(N01)
22|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1|case 6a||ZATU659x1(N01)||ZATU659|1(N01)
23|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 7b|SZITA~a1|case 6a||ZATU659x1(N01)||ZATU659|1(N01)
24|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 8b|SZITA~a1|case 6a||ZATU659x1(N01)||ZATU659|1(N01)
25|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 1b|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
26|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2b|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
27|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
28|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
29|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 5b|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
30|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
31|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2a|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
32|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
33|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
34|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
35|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 7b|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
36|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 8b|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
37|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 1b|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
38|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2b|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
39|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
40|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
41|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 5b|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
42|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
43|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2a|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
44|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
45|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
46|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
47|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 7b|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
48|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 8b|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
49|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 1b|SZITA~a1|case 2a||ZATU659x1(N01)||ZATU659|1(N01)
50|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2b|SZITA~a1|case 2a||ZATU659x1(N01)||ZATU659|1(N01)
51|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1|case 2a||ZATU659x1(N01)||ZATU659|1(N01)
52|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1|case 2a||ZATU659x1(N01)||ZATU659|1(N01)
53|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 5b|SZITA~a1|case 2a||ZATU659x1(N01)||ZATU659|1(N01)
54|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1|case 2a||ZATU659x1(N01)||ZATU659|1(N01)
55|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2a|SZITA~a1|case 2a||ZATU659x1(N01)||ZATU659|1(N01)
56|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1|case 2a||ZATU659x1(N01)||ZATU659|1(N01)
57|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1|case 2a||ZATU659x1(N01)||ZATU659|1(N01)
58|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1|case 2a||ZATU659x1(N01)||ZATU659|1(N01)
59|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 7b|SZITA~a1|case 2a||ZATU659x1(N01)||ZATU659|1(N01)
60|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 8b|SZITA~a1|case 2a||ZATU659x1(N01)||ZATU659|1(N01)
61|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 1b|SZITA~a1|case 7a||ZATU659x1(N01)||ZATU659|1(N01)
62|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2b|SZITA~a1|case 7a||ZATU659x1(N01)||ZATU659|1(N01)
63|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1|case 7a||ZATU659x1(N01)||ZATU659|1(N01)
64|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1|case 7a||ZATU659x1(N01)||ZATU659|1(N01)
65|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 5b|SZITA~a1|case 7a||ZATU659x1(N01)||ZATU659|1(N01)
66|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1|case 7a||ZATU659x1(N01)||ZATU659|1(N01)
67|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2a|SZITA~a1|case 7a||ZATU659x1(N01)||ZATU659|1(N01)
68|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1|case 7a||ZATU659x1(N01)||ZATU659|1(N01)
69|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1|case 7a||ZATU659x1(N01)||ZATU659|1(N01)
70|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1|case 7a||ZATU659x1(N01)||ZATU659|1(N01)
71|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 7b|SZITA~a1|case 7a||ZATU659x1(N01)||ZATU659|1(N01)
72|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 8b|SZITA~a1|case 7a||ZATU659x1(N01)||ZATU659|1(N01)
73|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 1b|SZITA~a1|case 8a||ZATU659x1(N01)||ZATU659|1(N01)
74|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2b|SZITA~a1|case 8a||ZATU659x1(N01)||ZATU659|1(N01)
75|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1|case 8a||ZATU659x1(N01)||ZATU659|1(N01)
76|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1|case 8a||ZATU659x1(N01)||ZATU659|1(N01)
77|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 5b|SZITA~a1|case 8a||ZATU659x1(N01)||ZATU659|1(N01)
78|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1|case 8a||ZATU659x1(N01)||ZATU659|1(N01)
79|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 2a|SZITA~a1|case 8a||ZATU659x1(N01)||ZATU659|1(N01)
80|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 3b|SZITA~a1|case 8a||ZATU659x1(N01)||ZATU659|1(N01)
81|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 4b|SZITA~a1|case 8a||ZATU659x1(N01)||ZATU659|1(N01)
82|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 6b|SZITA~a1|case 8a||ZATU659x1(N01)||ZATU659|1(N01)
83|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 7b|SZITA~a1|case 8a||ZATU659x1(N01)||ZATU659|1(N01)
84|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005213" title="to CDLI main page for this tablet">tablet P005213</a>|case 8b|SZITA~a1|case 8a||ZATU659x1(N01)||ZATU659|1(N01)
85|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">tablet P005214</a>|case 6b|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
86|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">tablet P005214</a>|case 3c|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
87|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">tablet P005214</a>|case 4c|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
88|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">tablet P005214</a>|case 5b|SZITA~a1|case 4a||ZATU659x1(N01)||ZATU659|1(N01)
89|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">tablet P005214</a>|case 6b|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
90|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">tablet P005214</a>|case 3c|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
91|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">tablet P005214</a>|case 4c|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
92|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005214" title="to CDLI main page for this tablet">tablet P005214</a>|case 5b|SZITA~a1|case 5a||ZATU659x1(N01)||ZATU659|1(N01)
93|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P005215" title="to CDLI main page for this tablet">tablet P005215</a>|case 1b|SZITA~a1|case 1a||ZATU659x1(N01)||ZATU659|1(N01)
94|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P002399" title="to CDLI main page for this tablet">tablet P002399</a>|case 2b|SZITA~a1|case 2a||ZATU659x1(N01)||ZATU659|1(N01)
95|<a target="_blank" href="https://cdli.ucla.edu/search/search_results.php?SearchMode=Text&ObjectID=P002399" title="to CDLI main page for this tablet">tablet P002399</a>|case 2b|SZITA~a1|case 3a||ZATU659x1(N01)||ZATU659|1(N01)

Now let's set up a class or group of orthographically related signs and search for this family of signs within a specific hiearchical level. There are two related signs in our key text, namely, SZITA~a1 and |SILA3~axKU6~a|, both of which include SILA3~a and refer to cuts of meat or fish.

Many of the related sign show up in butchering texts such as P000742, where there is a 20:1 ratio between all SILA3~a related signs and the number of sheep being butchered.

In [15]:
pNum = 'P000742'
query = f'''
tablet catalogId={pNum}
'''
results = A.search(query)
tablet = results[0][0]
A.lineart(tablet, width=400)
A.getSource(tablet)

1 result


['&P000742 = ATU 5, pl. 003, W 6066,a',
 '#atf: lang qpc ',
 '@obverse ',
 '@column 1 ',
 '1. 2(N14) , UDU~a ',
 '2. 2(N14) , SZITA~b1 ',
 '3. 2(N14) , BA 1(N57) SZITA~a1 ',
 '4. 8(N01) , KU3~a ',
 '5. 5(N34) 2(N14) , SZITA~a1 ',
 '6. 2(N14) , |(SZAxHI@g~a)~b| ',
 '7. 2(N14) , |(SZAxHI@g~a)~a| ',
 '@column 2 ',
 '1.a1. [1(N01)] , HI@g~a ',
 '1.a2. 1(N01) 1(N28) , ZATU714 ',
 '1.b. 2(N01) 1(N28) , |ZATU714xHI@g~a| MU ',
 '2. 3(N39~a) , HI@g~a SAG#? ',
 '3. , EN~a# SZE~a# [...] ',
 '4. , GESZTU~b ']

Here we have 20 sheep (UDU~a) and if we add up the numerals preceding the signs that include or derive from SILA3~a, namely SZITA~b1, SZITA~a1, |(SZAxHI@g~a)~b|, and |(SZAxHI@g~a)~a|, we arrive at a total of 400 (= 20 sheep times 20 cuts of meat per sheep).

Before doing the calculation, however, let's just list the signs.

In [16]:
A.lineart('SILA3~a')
A.lineart('SZITA~a1')
A.lineart('SZITA~b1')
A.lineart('SZAxHI@g~a)~b')
A.lineart('SZAxHI@g~a)~a')

In [17]:
sila3Related = dict(
    grapheme=SILA variant=3
    grapheme=SZITA variant=a1
)

SyntaxError: invalid syntax (<ipython-input-17-cf23b0a2839e>, line 2)

In [18]:
sila3RelatedPat = '|'.join(sila3Related)
query = f'''
tablet catalogId={pNum}
  face
    sign grapheme={sila3RelatedPat}
'''
results = A.search(query)
A.show(results)

NameError: name 'sila3Related' is not defined

## Checking our results with GREP

At least to some degree, the curious way in which subdivided lines and cases in proto-cuneiform are labeled in transliterations

* `1` is level 0
* `1a` is level 1
* `1a1` is level 2
* `1a1A` is level 3
* `1a1A1` is level 4

was originally meant to allow for a relatively easy way of isolating elements at a particular level using GREP. It is hard to grep for all cases of a certain level, because cases that are further subdivided also belong to the next lower level, so we can really only look at terminal cases at a certain level. 

So let's look for terminal cases at level 1. Text-fabric contends that there are 5468 terminal cases at level 1 and, in theory, a grep search that looks for lines beginning with a number, then a period, then a letter, then a period and a space, should bring up these 5468 terminal cases at level 1. The grep for this might look like the following:

```sh
cat uruk-*.txt | grep -c '^[0-9]\+\.\?[a-z]\+\.\? '
5431
```

In [5]:
query = '''
sign type=ideograph grapheme~[0-9]
'''
results = A.search(query)
print(f'{len(results)} results')
A.show(results, end=4)

14558 results
14558 results



##### Tablet 1



##### Tablet 2



##### Tablet 3



##### Tablet 4


Nearly right but we are missing line number that include a prime ('), so let's include that in the grep and try again:

```sh
cat uruk-*.txt | grep -c "^[0-9]\+\.\?[a-z]\+'\?\.\? "
5461
```

This is better, but grep is still missing 7 cases at level 1 that were identified by Text-Fabric. So let's print the grep results to a file and do the same with the Text-Fabric results, and then do a "diff," which compares the two text files and identifies differences.

```sh
cat uruk-*.txt | grep -n "^[0-9]\+\.\?[a-z]\+'\?\.\?[^0-9.]" > ~/Downloads/grep.txt
```

with open('/Users/dirk/Downloads/tf.txt', 'w') as fh:
    for c in A.casesByLevel(1):
        t = L.u(c, otype='tablet')[0]
        fh.write(f'{F.period.v(t)}.txt:{F.srcLnNum.v(c)}:{F.srcLn.v(c)}\n')

Here are the issues:

In the GREP results we see something that is missing in the TF results.
```
uruk-iii.txt:27736:70.a. ZATU786 BANSZUR~c 
uruk-iii.txt:27740:71.a. AD~b GISZ 
uruk-iii.txt:27752:76.a. GISZ KAK~a X 
uruk-iii.txt:27754:76.b. GISZ SI 
uruk-iii.txt:33185:49.a. [...] X GIR~c 
uruk-iii.txt:33187:49.b. NU GIR~c 
uruk-iii.txt:33203:56.a. |UR2+TAR| [...] 
uruk-iii.txt:33205:56.b. |UR2+TAR| AN TAR~a 
```

But this comes from a non-physical `P471694` which we have skipped when composing our corpus.

In the GREP results we see something else that is missing in the TF results.

```
uruk-iii.txt:75574:1.aB1. 2(N01) , SI4~f# 
uruk-iii.txt:75575:1.aB2. 1(N01) , SI# TUR# 
```

This is an unusual alteration from lowercase to uppercase in the numbering.
We have to reckon with it in our GREP pattern.

In the TF results, but not in the GREP results we see:
```
uruk-iii.txt:47635:2'.a. 1(N38) 
uruk-iii.txt:47636:2'.b. [...] , [...] 
```
We need to allow for primes in non-final places in our grep pattern.

We try again with this pattern:

```sh
grep -n "^[0-9]\+\'\?\.\?[a-z]\+\'\?\.\?[^0-9.A-Z]" uruk-*.txt > ~/Downloads/grep.txt
cat uruk-*.txt | grep -c "^[0-9]\+\'\?\.\?[a-z]\+\'\?\.\?[^0-9.A-Z]"
5476
```

Now the only differences between the GREP and TF results are those 8 lines, found by GREP in an excluded tablet. If we subtract that from 5476 we get 5468, exactly the amount reported by TF.

## Reflection on GREP

It is good to see that the GREP check passes. 
However, there are more excluded tablets that are seen by GREP and not by TF.
In this case, those have not level 1 subcases, as became clear when we diffed the GREP results with the TF results.

It is hard to GREP, because the ATF lines are overloaded with patterns that you need to remember.
If you forget some subtleties, you miss cases.

The TF data is much more straightforward, in that we can separate concerns. We do not have to remember primes, for example.

When we converted the ATF corpus into TF, we have dealt with all issues and checked them in the manner we have showed above, but much more comprehensively.

We trust that this exercise has raised your level of trust in the TF data.